merge feature/gefs_v12_3 into ops for GEFSv12.3.0

XianwuXue-NOAA opened the pull request.

XianwuXue-NOAA committed:
Improve pyGEFS and bin files for WCOSS2

 On branch feature/ops_rocoto_port2wcoss2
 Changes to be committed:
	modified:   rocoto/bin/wcoss2/archive_atm.sh
	modified:   rocoto/bin/wcoss2/archive_chem.sh
	modified:   rocoto/bin/wcoss2/archive_wave.sh
	modified:   rocoto/bin/wcoss2/atmos_post_cleanup.sh
	modified:   rocoto/bin/wcoss2/atmos_prep.sh
	modified:   rocoto/bin/wcoss2/avg_gempak_vgf.sh
	modified:   rocoto/bin/wcoss2/avgspr_gempak.sh
	modified:   rocoto/bin/wcoss2/avgspr_gempak_meta.sh
	modified:   rocoto/bin/wcoss2/chem_init.sh
	modified:   rocoto/bin/wcoss2/chem_post_cleanup.sh
	modified:   rocoto/bin/wcoss2/chem_prep_emissions.sh
	modified:   rocoto/bin/wcoss2/cleanup_atm.sh
	modified:   rocoto/bin/wcoss2/cleanup_chem.sh
	modified:   rocoto/bin/wcoss2/cleanup_wave.sh
	modified:   rocoto/bin/wcoss2/common.sh
	modified:   rocoto/bin/wcoss2/copy_init.sh
	modified:   rocoto/bin/wcoss2/cqpf.sh
	modified:   rocoto/bin/wcoss2/ensavg_nemsio.sh
	modified:   rocoto/bin/wcoss2/enspost.sh
	modified:   rocoto/bin/wcoss2/ensstat_hr.sh
	modified:   rocoto/bin/wcoss2/extractvars.sh
	modified:   rocoto/bin/wcoss2/fcst_post_manager.sh
	modified:   rocoto/bin/wcoss2/forecast_hr.sh
	modified:   rocoto/bin/wcoss2/gempak.sh
	modified:   rocoto/bin/wcoss2/gempak_meta.sh
	modified:   rocoto/bin/wcoss2/getcfssst.sh
	modified:   rocoto/bin/wcoss2/init_recenter.sh
	modified:   rocoto/bin/wcoss2/keep_data_atm.sh
	modified:   rocoto/bin/wcoss2/keep_data_chem.sh
	modified:   rocoto/bin/wcoss2/keep_data_wave.sh
	modified:   rocoto/bin/wcoss2/keep_init.sh
	modified:   rocoto/bin/wcoss2/post_genesis.sh
	modified:   rocoto/bin/wcoss2/post_hr.sh
	modified:   rocoto/bin/wcoss2/post_track.sh
	modified:   rocoto/bin/wcoss2/postsnd.sh
	modified:   rocoto/bin/wcoss2/prdgen_hr.sh
	modified:   rocoto/bin/wcoss2/wave_gempak.sh
	modified:   rocoto/bin/wcoss2/wave_init.sh
	modified:   rocoto/bin/wcoss2/wave_post.sh
	modified:   rocoto/bin/wcoss2/wave_prep.sh
	modified:   rocoto/bin/wcoss2/wave_stat.sh
	modified:   rocoto/py/GEFS_XML_For_Tasks.py
	modified:   rocoto/py/user_wcoss2.conf

Refs: #23

XianwuXue-NOAA committed:
Add nodesize to pyGEFS

 On branch feature/ops_rocoto_port2wcoss2
 Changes to be committed:
	modified:   rocoto/py/GEFS_XML_For_Tasks.py

Refs: #44

XianwuXue-NOAA committed:
Modify pyGEFS to run on WCOSS2

 On branch feature/ops_rocoto_port2wcoss2
 Changes to be committed:
	modified:   rocoto/compile_install_all.sh
	modified:   rocoto/parm/setbase
	modified:   rocoto/py/GEFS_UserConfig.py
	modified:   rocoto/py/GEFS_XML.py
	modified:   rocoto/py/GEFS_XML_For_Tasks.py

Refs: #44

XianwuXue-NOAA committed:
Change %J to %s for wcoss2 logfile

 On branch feature/ops_rocoto_port2wcoss2
 Changes to be committed:
	modified:   rocoto/py/user_wcoss2.conf

Refs: #44

XianwuXue-NOAA committed:
Improve bin files of first 3 tasks to WCOSS2

The 3 tasks are atmos_prep, getcfssst and init_recenter
    and also improve gefs_config for both wcoss2 and other machines

 On branch feature/ops_rocoto_port2wcoss2
 Changes to be committed:
	modified:   rocoto/bin/wcoss2/atmos_prep.sh
	modified:   rocoto/bin/wcoss2/common.sh
	modified:   rocoto/bin/wcoss2/getcfssst.sh
	modified:   rocoto/bin/wcoss2/init_recenter.sh
	modified:   rocoto/parm/gefs_config

Refs: #44

XianwuXue-NOAA committed:
Fix a bug

 On branch feature/ops_rocoto_port2wcoss2
 Changes to be committed:
	modified:   rocoto/py/GEFS_UserConfig.py

Refs: #44

XianwuXue-NOAA committed:
Modify bin files of rest of tasks for wcoss2

 On branch feature/ops_rocoto_port2wcoss2
 Changes to be committed:
	modified:   bin/wcoss2/*.sh

Refs: #46

XianwuXue-NOAA committed:
Improve pyGEFS to gen right resources for gempak
 On branch feature/ops_rocoto_port2wcoss2
 Changes to be committed:
	modified:   rocoto/py/GEFS_XML_For_Tasks.py

Refs: #44

XianwuXue-NOAA committed:
UPdate resources on wcoss2

 On branch feature/ops_rocoto_port2wcoss2
 Changes to be committed:
	modified:   rocoto/py/user_wcoss2.conf

Refs: #44

XianwuXue-NOAA committed:
Delete unused statements and script

 On branch feature/ops_rocoto_port2wcoss2
	modified:   rocoto/compile_install_all.sh
	modified:   rocoto/py/GEFS_XML.py
	deleted:    rocoto/py/run_to_get_all.py

Refs: #44

XianwuXue-NOAA committed:
Improve pyGEFS to generate resources for fcst

1) If node, ppt, tpp exists then use it first for fcst tasks
2) Modify resources of user_wcoss2.conf

 On branch feature/ops_rocoto_port2wcoss2
	modified:   rocoto/py/GEFS_XML_For_Tasks.py
	modified:   rocoto/py/user_wcoss2.conf

Refs: #44

XianwuXue-NOAA committed:
Fix some bugs

 On branch feature/ops_rocoto_port2wcoss2
 Changes to be committed:
	modified:   rocoto/py/GEFS_XML_For_Tasks.py

Refs: #44

XianwuXue-NOAA committed:
Make create_envar more flexiable

 On branch feature/ops_rocoto_port2wcoss2
 Changes to be committed:
	modified:   rocoto/py/GEFS_XML_For_Tasks.py

Refs: #44

XianwuXue-NOAA committed:
Fix a typo
 On branch feature/ops_rocoto_port2wcoss2
	modified:   rocoto/py/GEFS_XML_For_Tasks.py

Refs: #44

XianwuXue-NOAA committed:
Comment unused statement and add check_init_tpp

 On branch feature/ops_rocoto_port2wcoss2
 Changes to be committed:
	modified:   rocoto/py/GEFS_UserConfig.py
	modified:   rocoto/py/user_wcoss2.conf

Refs: #44

XianwuXue-NOAA committed:
Modify to port back to WCOSS1 for p35

 On branch feature/ops_rocoto_port2wcoss2_dell35

Refs: #48

XianwuXue-NOAA committed:
Add emc.ens to pyGEFS

 On branch feature/ops_rocoto_port2wcoss2
 Changes to be committed:
	modified:   rocoto/py/GEFS_XML.py

Refs: #44

XianwuXue-NOAA committed:
Improve user_full.conf to use default values

 On branch feature/ops_rocoto_port2wcoss2
 Changes to be committed:
	modified:   user_full.conf

Refs: #44

XianwuXue-NOAA committed:
Merge branch 'feature/ops_rocoto_port2wcoss2' of github.com:XianwuXue-NOAA/GEFS into feature/ops_rocoto_port2wcoss2

XianwuXue-NOAA committed:
Update HOMEDIR to use emc.ens on WCOSS2

 On branch feature/ops_rocoto_port2wcoss2
 Changes to be committed:
	modified:   parm/setbase
	modified:   user_full.conf

Refs: #44

XianwuXue-NOAA committed:
Merge branch 'feature/ops_rocoto_port2wcoss2' into feature/ops_rocoto_port2wcoss2_dell35

XianwuXue-NOAA committed:
Merge branch 'hotfix/ops' into feature/ops_rocoto_port2wcoss2_dell35

 Conflicts:
	jobs/JGEFS_ATMOS_INIT_RECENTER
	jobs/JGEFS_ATMOS_POST_CLEANUP
	jobs/JGEFS_ATMOS_PREP
	jobs/JGEFS_CHEM_INIT
	jobs/JGEFS_CHEM_PREP_EMISSIONS
	jobs/JGEFS_FORECAST

 On branch feature/ops_rocoto_port2wcoss2_dell35
 Changes to be committed:
	modified:   ecf/chem/jgefs_chem_forecast.ecf
	modified:   ecf/chem/jgefs_chem_init.ecf
	modified:   ecf/chem/jgefs_chem_post.ecf
	modified:   ecf/chem/jgefs_chem_prdgen.ecf
	modified:   ecf/chem/jgefs_chem_prep_emissions.ecf
	modified:   ecf/cleanup/jgefs_atmos_post_cleanup.ecf
	modified:   ecf/d0_16/atmos/jgefs_atmos_post.ecf
	modified:   ecf/d0_16/atmos/jgefs_atmos_postsnd.ecf
	modified:   ecf/d0_16/atmos/jgefs_atmos_prdgen.ecf
	modified:   ecf/d0_16/atmos/jgefs_atmos_prep.ecf
	modified:   ecf/d0_16/jgefs_fcst_post_manager.ecf
	modified:   ecf/d0_16/jgefs_forecast.ecf
	modified:   ecf/d0_16/wave/jgefs_wave_gempak.ecf
	modified:   ecf/d0_16/wave/jgefs_wave_post.ecf
	modified:   ecf/d0_16/wave/jgefs_wave_prep.ecf
	modified:   ecf/d16_35/atmos/jgefs_atmos_forecast.ecf
	modified:   ecf/d16_35/atmos/jgefs_atmos_post.ecf
	modified:   ecf/d16_35/atmos/jgefs_atmos_prdgen.ecf
	new file:   ecf/def/gefs_prod00.def
	new file:   ecf/def/gefs_prod06.def
	new file:   ecf/def/gefs_prod12.def
	new file:   ecf/def/gefs_prod18.def
	modified:   ecf/gempak/atmos/jgefs_atmos_avgspr_gempak_meta.ecf
	modified:   ecf/gempak/atmos/jgefs_atmos_gempak.ecf
	modified:   ecf/gempak/atmos/jgefs_atmos_gempak_meta.ecf
	modified:   ecf/init/atmos/jgefs_atmos_getcfssst.ecf
	modified:   ecf/init/atmos/jgefs_atmos_init_recenter.ecf
	modified:   ecf/init/wave/jgefs_wave_init.ecf
	modified:   ecf/post_processing/d0_16/atmos/jgefs_atmos_avg_postsnd.ecf
	modified:   ecf/post_processing/d0_16/atmos/jgefs_atmos_ensavg_nemsio.ecf
	modified:   ecf/post_processing/d0_16/atmos/jgefs_atmos_enspost.ecf
	modified:   ecf/post_processing/d0_16/atmos/jgefs_atmos_ensstat.ecf
	modified:   ecf/post_processing/d0_16/atmos/jgefs_atmos_prdgen_gfs.ecf
	modified:   ecf/post_processing/d0_16/wave/jgefs_wave_stat.ecf
	modified:   ecf/post_processing/d16_35/atmos/jgefs_atmos_enspost.ecf
	modified:   ecf/post_processing/d16_35/atmos/jgefs_atmos_ensstat.ecf
	modified:   jobs/JGEFS_ATMOS_AVGSPR_GEMPAK_META
	modified:   jobs/JGEFS_ATMOS_ENSAVG_NEMSIO
	modified:   jobs/JGEFS_ATMOS_ENSPOST
	modified:   jobs/JGEFS_ATMOS_ENSSTAT
	modified:   jobs/JGEFS_ATMOS_GEMPAK
	modified:   jobs/JGEFS_ATMOS_GEMPAK_META
	modified:   jobs/JGEFS_ATMOS_GETCFSSST
	modified:   jobs/JGEFS_ATMOS_INIT_RECENTER
	modified:   jobs/JGEFS_ATMOS_POST
	modified:   jobs/JGEFS_ATMOS_POSTSND
	modified:   jobs/JGEFS_ATMOS_POST_CLEANUP
	modified:   jobs/JGEFS_ATMOS_PRDGEN
	modified:   jobs/JGEFS_ATMOS_PREP
	modified:   jobs/JGEFS_CHEM_INIT
	modified:   jobs/JGEFS_CHEM_PREP_EMISSIONS
	modified:   jobs/JGEFS_FCST_POST_MANAGER
	modified:   jobs/JGEFS_FORECAST
	modified:   jobs/JGEFS_WAVE_GEMPAK
	modified:   jobs/JGEFS_WAVE_INIT
	modified:   jobs/JGEFS_WAVE_POST
	modified:   jobs/JGEFS_WAVE_PREP
	modified:   jobs/JGEFS_WAVE_STAT
	modified:   parm/gefs_ensstat.parm
	modified:   scripts/exgefs_chem_init.py
	modified:   sorc/README.build
	modified:   sorc/link_gefs.sh
	modified:   util/ush/recentensemble_prep.py
	modified:   versions/run.ver

Refs: #44

XianwuXue-NOAA committed:
Change KEEP_DIR from GEFS to GEFS_KEEP

 On branch feature/ops_rocoto_port2wcoss2_dell35
 Changes to be committed:
	modified:   py/user_cray.conf
	modified:   py/user_hera.conf
	modified:   py/user_wcoss2.conf
	modified:   py/user_wcoss_dell_p3.conf
	modified:   py/user_wcoss_dell_p35.conf

Refs: #44 and #48

XianwuXue-NOAA committed:
Port archive and keep_data to WCOSS2

However, further improvement of py/*.py to compatible with
    other machines is needed

 On branch feature/ops_rocoto_port2wcoss2_dell35
 Changes to be committed:
	modified:   rocoto/bin/py/archive_atm.py
	modified:   rocoto/bin/py/archive_chem.py
	modified:   rocoto/bin/py/archive_wave.py
	modified:   rocoto/bin/py/keep_data_atm.py
	modified:   rocoto/bin/py/keep_data_chem.py
	modified:   rocoto/bin/py/keep_data_wave.py
	modified:   rocoto/bin/wcoss2/archive_atm.sh
	modified:   rocoto/bin/wcoss2/archive_chem.sh
	modified:   rocoto/bin/wcoss2/archive_wave.sh
	modified:   rocoto/bin/wcoss2/keep_data_atm.sh
	modified:   rocoto/bin/wcoss2/keep_data_chem.sh
	modified:   rocoto/bin/wcoss2/keep_data_wave.sh

Refs: #44 and #49

XianwuXue-NOAA committed:
Fix some bugs in the scripts

 On branch feature/ops_rocoto_port2wcoss2_dell35
 Changes to be committed:
	modified:   jobs/JGEFS_ATMOS_INIT_RECENTER
	modified:   rocoto/bin/py/keep_data_wave.py
	modified:   scripts/exgefs_wave_gempak.sh

Refs: #44

XianwuXue-NOAA committed:
Add more data to DIRS_TO_ARCHIVE for WCOSS2

 On branch feature/ops_rocoto_port2wcoss2_dell35
 Changes to be committed:
	modified:   rocoto/py/user_wcoss2.conf

Refs: #44

XianwuXue-NOAA committed:
Add dep of prdgen_gfs on WCOSS2

 On branch feature/ops_rocoto_port2wcoss2_dell35
 Changes to be committed:
	modified:   rocoto/py/user_wcoss2.conf

Refs: #44

XianwuXue-NOAA committed:
Update the resources based on NCO's changes
 On branch feature/ops_rocoto_port2wcoss2_dell35
 Changes to be committed:
	modified:   rocoto/bin/wcoss2/wave_prep.sh
	modified:   rocoto/py/user_wcoss2.conf

Refs: #44

XianwuXue-NOAA committed:
Merge branch 'ops' into feature/ops_rocoto_port2wcoss2_dell35

 On branch feature/ops_rocoto_port2wcoss2_dell35
 Changes to be committed:
	modified:   ecf/def/gefs_prod00.def
	modified:   ecf/def/gefs_prod06.def
	modified:   ecf/def/gefs_prod12.def
	modified:   ecf/def/gefs_prod18.def
	modified:   jobs/JGEFS_ATMOS_GETCFSSST
	modified:   jobs/JGEFS_ATMOS_INIT_RECENTER
	modified:   jobs/JGEFS_WAVE_PREP
	modified:   parm/gefs_fcst.parm
	modified:   scripts/exgefs_wave_gempak.sh
	modified:   versions/run.ver

Refs: #44 and #49

XianwuXue-NOAA committed:
Merge pull request #5 from XianwuXue-NOAA/feature/ops_sync_NCO

Synchronize NCO's changes in prod on WCOSS2

XianwuXue-NOAA committed:
Modify compath to run realtime mode
    to read data from prod

 On branch feature/ops_rocoto_port2wcoss2_dell35
 Changes to be committed:
	modified:   rocoto/bin/wcoss2/common.sh

XianwuXue-NOAA committed:
Merge branch 'ops' into feature/ops_rocoto_port2wcoss2_dell35

XianwuXue-NOAA committed:
Change excl to exclhost based on NCO's changes

 On branch feature/ops_rocoto_port2wcoss2_dell35
 Changes to be committed:
	modified:   py/GEFS_XML_For_Tasks.py
	modified:   py/user_wcoss2.conf

Refs: #44

XianwuXue-NOAA committed:
Add rocotorun to compile_install_all.sh for WCOSS2

 On branch feature/ops_rocoto_port2wcoss2_dell35
 Changes to be committed:
	modified:   compile_install_all.sh

Refs: #44

XianwuXue-NOAA committed:
Merge branch 'feature/ops_sync_NCO' into feature/ops_rocoto_port2wcoss2_dell35

XianwuXue-NOAA committed:
Update DELTIM to 450 in all user_*.conf

 On branch feature/ops_rocoto_port2wcoss2_dell35
	modified:   rocoto/py/user_cray.conf
	modified:   rocoto/py/user_hera.conf
	modified:   rocoto/py/user_wcoss2.conf
	modified:   rocoto/py/user_wcoss_dell_p3.conf
	modified:   rocoto/py/user_wcoss_dell_p35.conf

XianwuXue-NOAA committed:
Update gempak_meta mem and dep of chem_init

 On branch feature/ops_rocoto_port2wcoss2_dell35
 Changes to be committed:
	modified:   rocoto/py/GEFS_XML_For_Tasks.py
	modified:   rocoto/py/user_wcoss2.conf
Refs: #44

XianwuXue-NOAA committed:
Merge branch 'feature/ops_sync_NCO' into feature/ops_rocoto_port2wcoss2_dell35

XianwuXue-NOAA committed:
Increase mems of prdgen_gfs and gempak_meta

To solve cgroup issue

 On branch feature/ops_rocoto_port2wcoss2_dell35
	modified:   rocoto/py/user_wcoss2.conf

Refs: #44

XianwuXue-NOAA committed:
Update the GEFS FIX and g-w check branch

 On branch feature/gefs_v12_3
 Changes to be committed:
	modified:   sorc/checkout.sh
	modified:   sorc/link_gefs.sh

Refs: #56

XianwuXue-NOAA committed:
Fix a bug for user_wcoss2.conf

 On branch feature/gefs_v12_3
	modified:   rocoto/py/user_wcoss2.conf

Refs: #44

XianwuXue-NOAA committed:
Port cleanup_atm[chem,wave] to wcoss2

 On branch feature/gefs_v12_3
	modified:   rocoto/bin/py/cleanup_atm.py
	modified:   rocoto/bin/py/cleanup_chem.py
	modified:   rocoto/bin/py/cleanup_wave.py
	modified:   rocoto/bin/wcoss2/cleanup_atm.sh
	modified:   rocoto/bin/wcoss2/cleanup_chem.sh
	modified:   rocoto/bin/wcoss2/cleanup_wave.sh

Refs: #44

XianwuXue-NOAA committed:
Improve to clean up the atm and chem data on WCOSS2
 On branch feature/gefs_v12_3
	modified:   rocoto/bin/py/cleanup_atm.py
	modified:   rocoto/bin/py/cleanup_chem.py

Refs: #44

XianwuXue-NOAA committed:
Merge branch 'ops' into feature/ops_rocoto_port2wcoss2_dell35

XianwuXue-NOAA committed:
Merge branch 'feature/ops_rocoto_port2wcoss2_dell35' into feature/gefs_v12_3

XianwuXue-NOAA commented on sorc/checkout.sh:
```diff
@@ -15,7 +15,7 @@ if [[ ! -d global-workflow.fd ]] ; then
     git clone https://github.com/NOAA-EMC/global-workflow.git global-workflow.fd >>  ${logs_dir}/checkout-global-workflow.log 2>&1
     cd global-workflow.fd
-    git checkout gefs_v12.2.0-1
+    git checkout release/gefs_v12 #gefs_v12.2.0-1
```
Will change the branch name to tag when the tag is available.

lipan-NOAA replied:
@XianwuXue-NOAA I agree

XianwuXue-NOAA committed:
Add three new variables AWC required

The new added variables are:
1) cloud ceiling (geopotential meters)
2) surface visibility (meters)
3) surface frozen precipitation fraction (percent)

 On branch feature/gefs_v12_3
	modified:   parm/gefs_pgrb2a_f00.parm_0p25
	modified:   parm/gefs_pgrb2a_fhh.parm_0p25

Refs: #63

XianwuXue-NOAA commented:
@WalterKolczynski-NOAA could you create a tag (gefs_v12.3.0-0?) in global-workflow for branch "release/gefs_v12" since Li and I have tested successfully and fixed the UPP issue and updated the UPP tag.

WalterKolczynski-NOAA commented:
Tag created: https://github.com/NOAA-EMC/global-workflow/releases/tag/gefs_v12.3.0-0

XianwuXue-NOAA commented:
Thanks @WalterKolczynski-NOAA

XianwuXue-NOAA committed:
Change checkout.sh to checkout tag gefs_v12.3.0-0
 On branch feature/gefs_v12_3
 Changes to be committed:
	modified:   sorc/checkout.sh

Refs: #56
